SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -1111,6 +1116,12 @@ - (void)sendMessageToRecipient:(OWSMessageSend *)messageSend
     BOOL isPoWRequired = YES; // TODO: Base on message type
     [[LokiAPI objc_sendSignalMessage:signalMessage to:recipient.recipientId timestamp:message.timestamp requiringPoW:isPoWRequired]
         .thenOn(OWSDispatch.sendingQueue, ^(id result) {
+
+            // Update the thread's friend request status if needed
+            if (messageType == TSFriendRequestMessageType) {
+                message.thread.friendRequestStatus = TSThreadFriendRequestStatusSent;
```

Comment:
thread needs to be saved here
`[message.thread save];`

Reply:
We also need to set the status to `None` if an error occurs.

Reply:
Yeah I know, this was just a WIP commit :)